Services/UserService/UserService.cs

```diff
@@ -0,0 +1,141 @@
+﻿using AutoMapper;
+using webapi.Dtos.Users;
+using webapi.Models;
+
+namespace webapi.Services.UserService
+{
+    /// <summary>
+    /// Implementation of the user service.
+    /// </summary>
+    public class UserService : IUserService
+    {
+        // In-memory storage for users
+        private static List<Users> users = new List<Users>();
+
+        private readonly IMapper _mapper;
+
+        public UserService(IMapper mapper)
+        {
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Retrieves all users.
+        /// </summary>
+        /// <returns>A list of all users.</returns>
+        public async Task<List<UsersDto>> GetAllUsers()
+        {
+            return (users.Select(c => _mapper.Map<UsersDto>(c))).ToList();
+        }
+
+        /// <summary>
+        /// Retrieves a user by ID.
+        /// </summary>
+        /// <param name="id">The ID of the user to retrieve.</param>
+        /// <returns>The user with the specified ID, or null if not found.</returns>
+        public async Task<UsersDto> GetUserById(int id)
+        {
+            try
+            {
+                UsersDto result = _mapper.Map<UsersDto>(users.FirstOrDefault(c => c.Id == id));
+                if (result != null)
+                {
+                    return result;
+                }
+                else
+                {
+                    throw new Exception("User not found.");
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Adds a new user.
+        /// </summary>
+        /// <param name="user">The user to add.</param>
+        /// <returns>A list of all users including the newly added user.</returns>
+        public async Task<List<UsersDto>> AddUser(UsersDto newUser)
+        {
+            Users user = _mapper.Map<Users>(newUser);
+
+            // Assign a unique ID to the user
+            if (users.Any())
+            {
+                user.Id = users.Max(u => u.Id) + 1;
+            }
+            else
+            {
+                user.Id = 1;
+            }
+
+            // Add the user to the in-memory storage
+            users.Add(user);
+
+            return (users.Select(u => _mapper.Map<UsersDto>(u))).ToList();
```

Comment:
why are you sending password field also.

Reply:
done